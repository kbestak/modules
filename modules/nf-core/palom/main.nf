process PALOM {
    tag   "$meta.id"
    label 'process_single'

    container "docker.io/kbestak/palom:2024.10.2"

    input:
    tuple val(meta), path(images), val(pixel_size)

    output:
    tuple val(meta), path("*.ome.tif"), emit: registered_image
    path "versions.yml"               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "Conda not yet supported. Please use Docker / Singularity instead."
    }
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    python3.7 /work/align_multiple_cycles.py \\
        --img_list ${images} \\
        --out_dir . \\
        --out_name ${prefix}.ome.tif \\
        --px_size ${pixel_size} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        palom: \$( python -m pip show --version palom | grep "Version" | sed -e "s/Version: //g" )
        python: \$( python --version | sed -e "s/Python //g" )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.ome.tif

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        palom: \$( python -m pip show --version palom | grep "Version" | sed -e "s/Version: //g" )
        python: \$( python --version | sed -e "s/Python //g" )
    END_VERSIONS
    """
}
