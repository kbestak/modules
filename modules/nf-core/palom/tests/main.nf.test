nextflow_process {

    name "Test Process PALOM"
    script "../main.nf"
    config "./nextflow.config"
    process "PALOM"

    tag "modules"
    tag "modules_vib"
    tag "palom"

    test("palom - tiff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file('../../../../testdata/2channel_reference.tiff', checkIfExists: true),
                        file('../../../../testdata/2channel_moving.tiff', checkIfExists: true)
                    ],
                    0.23
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("palom - tiff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file('../../../../testdata/2channel_reference.tiff', checkIfExists: true),
                        file('../../../../testdata/2channel_moving.tiff', checkIfExists: true)
                    ],
                    0.23
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
