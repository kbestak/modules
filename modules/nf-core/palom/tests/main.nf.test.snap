{
    "palom - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.ome.tif:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d157b6c8b22f7304d0a7b3a5b790c85d"
                ],
                "registered_image": [
                    [
                        {
                            "id": "test"
                        },
                        "test.ome.tif:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d157b6c8b22f7304d0a7b3a5b790c85d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-05T12:53:20.18598769"
    },
    "palom - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.ome.tif:md5,c714732451d3cc6113b3f0717574c8a3"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d157b6c8b22f7304d0a7b3a5b790c85d"
                ],
                "registered_image": [
                    [
                        {
                            "id": "test"
                        },
                        "test.ome.tif:md5,c714732451d3cc6113b3f0717574c8a3"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d157b6c8b22f7304d0a7b3a5b790c85d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-05T12:54:49.087926853"
    }
}